test/unit_tests/usecase/get_cards_usecase_test.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:hearthstone_cards/src/core/parameter/card_request_params.dart';
+import 'package:hearthstone_cards/src/core/usecases/usecase.dart';
+import 'package:hearthstone_cards/src/core/util/constants.dart';
+import 'package:hearthstone_cards/src/data/model/card_model.dart';
+import 'package:hearthstone_cards/src/domain/repository/card_repository.dart';
+import 'package:hearthstone_cards/src/domain/usecase/get_cards_usecase.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../../mocks.dart';
+import 'get_cards_usecase_test.mocks.dart';
+
+@GenerateMocks([CardRepository])
+void main (){
+  late MockCardRepository cardRepository;
+  late CardsRequestParams params;
+  late Usecase getCardsUsecase;
+
+  setUp(() {
+    cardRepository = MockCardRepository();
+    params = CardsRequestParams();
+    getCardsUsecase = GetCardsUsecase(cardRepository);
+  });
+
+  group("Get Cards Usecase Unit Tests", () {
+    test("Call method without endpoint", () async {
+      List<CardModel> cardModelList = [CardModel.fromJson(cardsTestsJson)];
+      when(cardRepository.fetchAllCards(params)).thenAnswer((_) async{
+        return cardModelList;
+      });
+      expectLater(await getCardsUsecase(params: params), isA<List<CardModel>>());
+    });
+    test("Call method with endpoint", () async {
+      List<CardModel> cardModelList = [CardModel.fromJson(cardsTestsJson)];
+      params.endpoint = Constants.apiCardsQualityEndpoint["Free"];
+      when(cardRepository.fetchFilteredCards(params)).thenAnswer((_) async{
+        return cardModelList;
+      });
+      expectLater(await getCardsUsecase(params: params), isA<List<CardModel>>());
+    });
+  });
+
+
+
+}
```

Comment:
Remove this extras lines and add new line at EOF

Reply:
Done